src/scheduler/executionHistory.tsx

```diff
@@ -0,0 +1,219 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { PickersDayProps, PickersDay } from '@mui/x-date-pickers/PickersDay';
+import { DateCalendar } from '@mui/x-date-pickers/DateCalendar';
+import dayjs, { Dayjs } from 'dayjs';
+import ListDagRuns from './listDagRuns';
+import { LabIcon } from '@jupyterlab/ui-components';
+import LeftArrowIcon from '../../style/icons/left_arrow_icon.svg';
+import ListDagTaskInstances from './listDagTaskInstances';
+
+const iconLeftArrow = new LabIcon({
+  name: 'launcher:left-arrow-icon',
+  svgstr: LeftArrowIcon
+});
+
+const ExecutionHistory = ({
+  composerName,
+  dagId,
+  handleBackButton
+}: {
+  composerName: string;
+  dagId: string;
+  handleBackButton: () => void;
+}): JSX.Element => {
+  const [dagRunId, setDagRunId] = useState('');
+  const currentDate = new Date().toLocaleDateString();
+  const [selectedDate, setSelectedDate] = useState<Dayjs | null>(null);
+  const [startDate, setStartDate] = useState('');
+  const [endDate, setEndDate] = useState('');
+
+  const [blueListDates, setBlueListDates] = useState<string[]>([]);
+  const [greyListDates, setGreyListDates] = useState<string[]>([]);
+  const [orangeListDates, setOrangeListDates] = useState<string[]>([]);
+  const [redListDates, setRedListDates] = useState<string[]>([]);
+  const [greenListDates, setGreenListDates] = useState<string[]>([]);
+  const [darkGreenListDates, setDarkGreenListDates] = useState<string[]>([]);
+
+  const handleMonthChange = () => {
+    setDagRunId('');
+    setSelectedDate(null);
+  };
+
+  const handleDateSelection = (selectedValue: any) => {
+    setDagRunId('');
+    setSelectedDate(selectedValue);
+  };
+
+  const CustomDay = (props: PickersDayProps<Dayjs>) => {
+    const { day, isFirstVisibleCell, isLastVisibleCell } = props;
+    if (isFirstVisibleCell) {
+      setStartDate(new Date(day.toDate()).toISOString());
```

Comment:
Few things wrong here.
1) you shouldn't call a setstate method in the render method
2) you are calling the setstate method of ExecutionHistory.

Plase move customday to a toplevel component and cleanup the data flow

Reply:
Need start date and end date values from MUI calendar for current active month - Used for Dag Run API call from FE